src/App.tsx

```diff
@@ -15,47 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  let copyGoods = [...goodsFromServer];
+  const [sortType, setsortType] = useState(SortType.NONE);
+  const [isReversed, setReversed] = useState(false);
+
+  copyGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    copyGoods.reverse();
+  }
+
+  const sortAlphabetically = () => {
+    setsortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setsortType(SortType.LENGTH);
+  };
+
+  const reverce = () => {
```

Comment:
```suggestion
  const reverse = () => {
```
Btw, it's called the same as array's method, so it would be better to choose a bit another name